### PR TITLE
Support KSS on simulation mode

### PR DESCRIPTION
**Pull request by volcano0dr**
issue: #883 
Signed-off-by: volcano0dr <volcano_dr@163.com>

### CODE REVIEW

File: sdk/simulation/tinst/t_instructions.cpp [hunk 1]
```diff
@@ -207,6 +237,34 @@ static int _EGETKEY(sgx_key_request_t* kr, sgx_key_128bit_t okey)
...
+        if (kr->key_policy & SGX_KEYPOLICY_CONFIGID) {
+            dd.ddpsk.config_svn = kr->config_svn;
+            memcpy(&dd.ddpsk.config_id, &cur_secs->config_id, sizeof(sgx_config_id_t));
+        }
+
+        memcpy(&dd.ddpsk.tmp_attr, &tmp_attr, sizeof(sgx_attributes_t));
+        memcpy(&dd.ddpsk.attribute_mask, &kr->attribute_mask, sizeof(sgx_attributes_t));
+        memcpy(&dd.ddpsk.cpu_svn,&kr->cpu_svn,sizeof(sgx_cpu_svn_t));
+        dd.ddpsk.isv_svn = kr->isv_svn;
+        dd.ddpsk.isv_prod_id = cur_secs->isv_prod_id;
```
Comment (lzha101) [inline sdk/simulation/tinst/t_instructions.cpp:265]: Also need to check if (!(kr->key_policy & SGX_KEYPOLICY_NOISVPRODID)) before set isv_prod_id.

File: sdk/simulation/tinst/deriv.h [hunk 2]
```diff
@@ -47,16 +47,20 @@ typedef uint8_t se_owner_epoch_t[OWNEREPOCH_SIZE];
...
+    sgx_attributes_t     attribute_mask;  /* attribute mask from KEYREQUEST */
+    se_owner_epoch_t     csr_owner_epoch;
+    sgx_cpu_svn_t        cpu_svn;         /* CPUSVN from KEYREQUEST */
+    sgx_isv_svn_t        isv_svn;         /* ISVSVN from KEYREQUEST */
+    sgx_prod_id_t        isv_prod_id;     /* ISV PRODID from SECS   */
+    sgx_config_svn_t     config_svn;      /* CONFIGSVN from KEYREQUEST */
+    sgx_config_id_t      config_id;       /* CONFIGID from SECS      */
+    sgx_isvfamily_id_t   isv_family_id;   /* ISV FAMILYID from SECS  */
+    sgx_isvext_prod_id_t isv_ext_prod_id; /* ISV EXTPRODID from SECS */
+    sgx_measurement_t    mrenclave;
```
Comment (lzha101) [inline sdk/simulation/tinst/deriv.h:61]: Our previous derived data are based on SGX1.  From latest SDM, the derived data also includes some other fields, such as misc_select/misc_mask, key_policy.  Do you want to update this to more align with the latest SDM?

File: sdk/simulation/tinst/t_instructions.cpp (hunk 1 above)
Comment (volcano0dr) [reply sdk/simulation/tinst/t_instructions.cpp]: Done. Thanks for your review.

File: sdk/simulation/tinst/deriv.h (hunk 2 above)
Comment (volcano0dr) [reply sdk/simulation/tinst/deriv.h:64]: I have updated the derived data based on the latest SDM.

**Pull request merged**